R/utils.R

```diff
@@ -18,12 +18,13 @@
 add_partial_keys <- function(db_data,
                              var = NULL) {
   if (!is.null(enexpr(var))) {
-    pattern <- "^(\\w+?)_arm_(\\d)$"
+    # Include handling for instances where REDCap appends with "_1b" or similar
+    pattern <- "^(\\w+?)_arm_(\\d+\\w?)$"
 
     db_data <- db_data %>%
       mutate(
         redcap_event = sub(pattern, "\\1", {{ var }}),
-        redcap_arm   = as.integer(sub(pattern, "\\2", {{ var }}))
+        redcap_arm   = as.factor(sub(pattern, "\\2", {{ var }}))
```

Comment:
Convert to factor to capture non-integer arm values as mentioned in #206

Reply:
I think `character` is probably better here since these will usually be numbers and aren't the type of categorical variables you normally expect as factor to represent